### ISSUE TITLE
Failed to list GitHub repositories for search

### ISSUE THREAD

**Issue by chrismwendt** [sourcegraph/sourcegraph#4040]
I spun up a v3.4.0-rc.2 instance, added an external service, then updated it to add `sourcegraph/sourcegraph` and got this error:

> Error saving invalid configuration:

> warning: external service updated, but sync request failed: syncer.sync.sourced: 1 error occurred:
> * failed to list GitHub repositories for search: page='\x02', searchString="sourcegraph/sourcegraph",: github repository search returned incomplete results. This is an ephemeral error: query="sourcegraph/sourcegraph" page=2 total=418

![image](https://user-images.githubusercontent.com/1387653/57799217-6daa5f00-7703-11e9-94a1-8ea4d373c7ca.png)

- I don't think this configuration is invalid
- It says "invalid configuration", which seems to contradict "external service updated"
- Why is it running a search? I specifically asked for "sourcegraph/sourcegraph"
- The error seems rather cryptic to show to a user
- I don't know what to do to fix this (maybe the error message could state some action that I can take?)

cc @sourcegraph/core-services 

Similar to https://github.com/sourcegraph/sourcegraph/issues/4052

**Comment by chrismwendt** [comment]
Oh, the repository `sourcegraph/sourcegraph` goes under the `repos` field that I didn't have.

**My thought process** I realized I was wrong after being surprised when I saw that a ton of repositories were added. I guessed it had added all repositories affiliated with my token, which was also wrong. Then I remembered I hadn't set `affiliated` in the config, and the next thought was that it was running a GitHub repository search for the text `sourcegraph/sourcegraph`. Then I deduced that `repositoryQuery` must not be the right place for `sourcegraph/sourcegraph`, at which point I pulled up the docs again and realized that I had mistaken the `repos` field in the docs for `repositoryQuery`. Later I saw that there's a specific button to add a single repository, and I probably didn't do that initially because I thought I knew how to configure the external service.

I'll leave this issue open because I think other users might hit this snag, too.
